go/proxyd/server.go

```diff
@@ -162,6 +173,12 @@ func (s *Server) HandleRPC(w http.ResponseWriter, r *http.Request) {
 		batchRes := make([]*RPCRes, len(reqs))
 		var batchContainsCached bool
 		for i := 0; i < len(reqs); i++ {
+			if ctx.Err() == context.DeadlineExceeded {
```

Comment:
Should increment a metric when this happens - `proxyd_short_circuited_batches` or something.